pkg/controller/deployment_recreate.go

```diff
@@ -25,10 +25,18 @@ package controller
 import (
 	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
 )
 
 // rolloutRecreate implements the logic for recreating a machine set.
 func (dc *controller) rolloutRecreate(d *v1alpha1.MachineDeployment, isList []*v1alpha1.MachineSet, machineMap map[types.UID]*v1alpha1.MachineList) error {
+
+	clusterAutoscalerScaleDownAnnotations := make(map[string]string)
+	clusterAutoscalerScaleDownAnnotations[ClusterAutoscalerScaleDownDisabledAnnotationKey] = ClusterAutoscalerScaleDownDisabledAnnotationValue
+
+	// We do this to avoid accidentally deleting the user provided annotations.
+	clusterAutoscalerScaleDownAnnotations[ClusterAutoscalerScaleDownDisabledAnnotationByMCMKey] = ClusterAutoscalerScaleDownDisabledAnnotationByMCMValue
```

Comment:
We add this annotation in both cases, right? i.e When no such annotation is added by the user and when it is added by the user. How are we keeping track of when the annotation was added by user prior to adding of this annotation during rollout. Maybe I missed it in the changes?

Reply:
It is not handled while adding the annotation but handled during [removal](https://github.com/gardener/machine-controller-manager/pull/496/files#diff-d60d538aae4eb1c57f7cd31e87804382R461).

Reply:
Yes, basically we need a way for controller to identify if the existing ca-annotation was added by the user or mcm. 
- If the `by-mcm` annotation exists, MCM assumes, it is added by the MCM and removes it.
- In the next iteration, the user-initiated annotations will be put again, maybe via `NodeTemplate`. And not MCM won't remove it, as `by-mcm` annotation is missing.
- During the roll-out, we add the ca-annotations on all nodes, irrespective of they exist there already or not.

Reply:
Okay yes, forgot about the `NodeTemplate` readdition.